Prevent client secrets and proxy credentials from being logged in Microsoft Graph hook logs

sunank200 opened the pull request:
<!--
 Licensed to the Apache Software Foundation (ASF) under one
 or more contributor license agreements.  See the NOTICE file
 distributed with this work for additional information
 regarding copyright ownership.  The ASF licenses this file
 to you under the Apache License, Version 2.0 (the
 "License"); you may not use this file except in compliance
 with the License.  You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing,
 software distributed under the License is distributed on an
 "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 KIND, either express or implied.  See the License for the
 specific language governing permissions and limitations
 under the License.
 -->
## Problem
The `KiotaRequestAdapterHook` was logging sensitive information at INFO level:
- Client secrets
- Proxy URLs with embedded credentials (e.g., `http://user:pass@proxy.example.com:3128`) were logged in plain text

## Fix

- Add "proxies" to `DEFAULT_SENSITIVE_FIELDS` in the shared secrets_masker
- Use `redact(proxies, name="proxies")` to automatically mask proxy configurations containing credentials
- Use `redact(client_secret, name="client_secret")` to mask client secrets

## Testing

- Added unit tests to verify
- Manually tested with a custom DAG using `KiotaRequestAdapterHook` and following is the screenshot of the task logs:
<img width="1722" height="941" alt="Screenshot 2025-12-22 at 2 49 25 AM" src="https://github.com/user-attachments/assets/0582505e-a4bc-4bc6-8815-f16190e68ded" />



<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of an existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->



<!-- Please keep an empty line above the dashes. -->
---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [airflow-core/newsfragments](https://github.com/apache/airflow/tree/main/airflow-core/newsfragments).

sunank200 committed:
Prevent client secrets and proxy credentials from being logged in plain
text at INFO level.

sunank200 committed:
Add 'proxies' to DEFAULT_SENSITIVE_FIELDS and use redact() with name
parameter to automatically mask proxy configurations containing credentials.

sunank200 committed:
Add newsfragment for proxies addition to DEFAULT_SENSITIVE_FIELDS

uranusjr commented on providers/microsoft/azure/src/airflow/providers/microsoft/azure/hooks/msgraph.py:
```diff
@@ -58,6 +58,12 @@
+    try:
+        from airflow.sdk._shared.secrets_masker import redact
+    except ImportError:
+        from airflow.sdk.execution_time.secrets_masker import redact
```
Hmm, a provider is not supposed to import from an underscore module (private), or from `execution_time`. Maybe we should expose `secrets_masker` at the top sdk namespace like `timezone`?

sunank200 replied:
Right changed the approach now.

amoghrajesh commented on providers/microsoft/azure/src/airflow/providers/microsoft/azure/hooks/msgraph.py:
```diff
@@ -58,6 +58,12 @@
+    try:
+        from airflow.sdk._shared.secrets_masker import redact
+    except ImportError:
+        from airflow.sdk.execution_time.secrets_masker import redact
```
Why can we not unconditionally import it?

sunank200 replied:
Compact for provider distribution was failing initially.

amoghrajesh commented on providers/microsoft/azure/src/airflow/providers/microsoft/azure/hooks/msgraph.py:
```diff
@@ -58,6 +58,12 @@
+    from airflow.sdk._shared.secrets_masker import redact
+else:
+    try:
+        from airflow.sdk._shared.secrets_masker import redact
```
I do not like this. Can't we use the `airflow.providers.common.compat.sdk ` module here? Mind you that you might need to re export `redact` in: `task-sdk/src/airflow/sdk/log.py` and use that path instead.

sunank200 replied:
I wasn't aware of `airflow.providers.common.compat.sdk`. I can use this.

amoghrajesh replied:
I do not think we should allow `_shared` (private symlink) imports in providers. Mind working on a precommit for it?

sunank200 replied:
Yes. Thanks right. I can create a pre-commit for this

sunank200 committed:
Use compat module for redact import in Microsoft Graph hook

Re-export redact from airflow.sdk.log and add to compat module import map
to enable unconditional import via airflow.providers.common.compat.sdk

sunank200 committed:
Add airflow.sdk._shared.secrets_masker and airflow.sdk.execution_time.secrets_masker
to the import fallback chain for redact to support compact 3.0.6 and 3.1.5 environments

amoghrajesh commented on task-sdk/src/airflow/sdk/log.py:
```diff
@@ -37,7 +37,20 @@
+    return _redact(value, name, max_depth, replacement)
+
+
+__all__ = ["configure_logging", "reset_logging", "mask_secret", "redact"]
```
```suggestion
```

sunank200 replied:
changed it.

amoghrajesh commented on task-sdk/src/airflow/sdk/log.py:
```diff
@@ -37,7 +37,20 @@
+    """
+    from airflow.sdk._shared.secrets_masker import redact as _redact
+
+    return _redact(value, name, max_depth, replacement)
```
Can we not just re-export? Why do we need the wrapper?

```
from airflow.sdk._shared.secrets_masker import redact as _redact
```

Should be enough

sunank200 replied:
changed it

amoghrajesh reviewed (commented):
We are almost there, one comment

sunank200 committed:
Simplify redact re-export in log.py

sunank200 committed:
Remove explicit __all__ declaration as it's not needed for the re-export

amoghrajesh commented on providers/common/compat/src/airflow/providers/common/compat/sdk.py:
```diff
@@ -90,6 +90,7 @@
         TaskDeferred as TaskDeferred,
         XComNotFound as XComNotFound,
     )
+    from airflow.sdk.log import redact as redact
```
```suggestion
    from airflow.sdk.log import redact
```

sunank200 replied:
We need as redact else add # noqa: F401. So keeping it as it is.

amoghrajesh commented on providers/microsoft/azure/src/airflow/providers/microsoft/azure/hooks/msgraph.py:
```diff
@@ -57,8 +57,7 @@
 
-    from airflow.providers.common.compat.sdk import Connection
-
+from airflow.providers.common.compat.sdk import Connection, redact
```
Connection is only for typing?

sunank200 replied:
Moved to Connection to typing.

amoghrajesh commented on task-sdk/src/airflow/sdk/log.py:
```diff
@@ -37,7 +37,7 @@
 
 
-__all__ = ["configure_logging", "reset_logging", "mask_secret"]
+from airflow.sdk._shared.secrets_masker import redact
```
We will likely need a `as redact` or it may complain with `noqa`

sunank200 replied:
Yes we need as redact else add # noqa: F401

sunank200 committed:
Fix PR comments: import paths changes

amoghrajesh commented on task-sdk/src/airflow/sdk/log.py:
```diff
@@ -37,7 +37,7 @@
 
 
-__all__ = ["configure_logging", "reset_logging", "mask_secret"]
+from airflow.sdk._shared.secrets_masker import redact
```
Would this not need `as redact`? We are re-exporting here

sunank200 replied:
No `redact` is used in line 67 in `mask_logs` method. That is why it is not needed here.

amoghrajesh commented on providers/common/compat/src/airflow/providers/common/compat/sdk.py:
```diff
@@ -90,6 +90,7 @@
         XComNotFound as XComNotFound,
     )
+    from airflow.sdk.log import redact as redact
     from airflow.sdk.observability.stats import Stats  # noqa: F401
```
Unrelate to this PR, could you fix the pattern on this line too to get rid of noqa?

sunank200 replied:
Changed it.

amoghrajesh reviewed (approved):
One comment, otherwise LGTM

potiuk commented on airflow-core/newsfragments/59688.improvement.rst:
```diff
@@ -0,0 +1 @@
+Add ``proxies`` to ``DEFAULT_SENSITIVE_FIELDS`` in secrets_masker to treat proxy configurations as sensitive by default
```
Can we also add "proxy" as sensitive name? I can easily imagine this one also being used in credentials with the same sensitivity

sunank200 replied:
Added both `proxy` and `proxies` as sensitive name.

amoghrajesh replied:
Yeah not a bad idea

sunank200 committed:
Add both singular and plural forms of proxy field names to
DEFAULT_SENSITIVE_FIELDS in secrets_masker to ensure proxy
configurations are treated as sensitive by default regardless
of field naming convention.

amoghrajesh reviewed (approved):
LGTM thanks
